File: python/newnanmanager/models/player.py
"""Player-related data models."""

from typing import Optional

from pydantic import BaseModel, Field

from .common import PagedData
from .enums import BanMode


class Player(BaseModel):
    """玩家实体."""

    id: int = Field(description="玩家ID")
    name: str = Field(description="游戏ID")
    town_id: Optional[int] = Field(default=None, description="所属城镇ID")
    qq: Optional[str] = Field(default=None, description="QQ号")
    qqguild: Optional[str] = Field(default=None, description="QQ频道ID")
    discord: Optional[str] = Field(default=None, description="Discord ID")
    in_qq_group: bool = Field(description="是否在QQ群")
    in_qq_guild: bool = Field(description="是否在QQ频道")
    in_discord: bool = Field(description="是否在Discord群")
    ban_mode: BanMode = Field(description="封禁模式")
    ban_expire: Optional[str] = Field(
        default=None, description="封禁到期时间(ISO8601格式)"
    )
    ban_reason: Optional[str] = Field(default=None, description="封禁原因")
    created_at: str = Field(description="创建时间(ISO8601格式)")
    updated_at: str = Field(description="更新时间(ISO8601格式)")


class PlayersListData(PagedData[Player]):
    """玩家列表数据."""

    players: list[Player] = Field(default_factory=list, description="玩家列表")

    @property
    def items(self) -> list[Player]:
        """获取数据项列表."""
        return self.players

    @items.setter
    def items(self, value: list[Player]) -> None:
        """设置数据项列表."""
        self.players = value


class PlayerLoginInfo(BaseModel):
    """玩家登录信息."""

    player_id: int = Field(description="玩家ID")
    name: str = Field(description="玩家名")
    ip: str = Field(description="登录IP")


class PlayerValidateInfo(BaseModel):
    """单个玩家验证信息."""

    player_name: str = Field(description="玩家名")
    ip: str = Field(description="IP地址")
    client_version: Optional[str] = Field(default=None, description="客户端版本")
    protocol_version: Optional[str] = Field(default=None, description="协议版本")


class PlayerValidateResult(BaseModel):
    """单个玩家验证结果."""

    player_name: str = Field(description="玩家名")
    allowed: bool = Field(description="是否允许登录")
    player_id: Optional[int] = Field(default=None, description="玩家ID")
    reason: Optional[str] = Field(default=None, description="拒绝原因")
    newbie: bool = Field(default=False, description="是否为新玩家")
    ban_mode: Optional[BanMode] = Field(default=None, description="封禁模式")
    ban_expire: Optional[str] = Field(
        default=None, description="封禁到期时间(ISO8601格式)"
    )
    ban_reason: Optional[str] = Field(default=None, description="封禁原因")


class ValidateData(BaseModel):
    """玩家验证响应数据."""

    results: list[PlayerValidateResult] = Field(description="验证结果列表")
    processed_at: int = Field(description="处理时间戳")

File: csharp/Models/Player.cs
using System.Text.Json.Serialization;

namespace NewNanManager.Client.Models;

/// <summary>
/// 玩家实体
/// </summary>
public class Player
{
    /// <summary>
    /// 玩家ID
    /// </summary>
    [JsonPropertyName("id")]
    public int Id { get; set; }

    /// <summary>
    /// 玩家名称
    /// </summary>
    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 所属城镇ID
    /// </summary>
    [JsonPropertyName("town_id")]
    public int? TownId { get; set; }

    /// <summary>
    /// QQ号
    /// </summary>
    [JsonPropertyName("qq")]
    public string? QQ { get; set; }

    /// <summary>
    /// QQ频道
    /// </summary>
    [JsonPropertyName("qqguild")]
    public string? QQGuild { get; set; }

    /// <summary>
    /// Discord
    /// </summary>
    [JsonPropertyName("discord")]
    public string? Discord { get; set; }

    /// <summary>
    /// 是否在QQ群中
    /// </summary>
    [JsonPropertyName("in_qq_group")]
    public bool InQQGroup { get; set; }

    /// <summary>
    /// 是否在QQ频道中
    /// </summary>
    [JsonPropertyName("in_qq_guild")]
    public bool InQQGuild { get; set; }

    /// <summary>
    /// 是否在Discord中
    /// </summary>
    [JsonPropertyName("in_discord")]
    public bool InDiscord { get; set; }

    /// <summary>
    /// 封禁模式
    /// </summary>
    [JsonPropertyName("ban_mode")]
    public BanMode BanMode { get; set; }

    /// <summary>
    /// 封禁过期时间
    /// </summary>
    [JsonPropertyName("ban_expire")]
    public DateTime? BanExpire { get; set; }

    /// <summary>
    /// 封禁原因
    /// </summary>
    [JsonPropertyName("ban_reason")]
    public string? BanReason { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    [JsonPropertyName("created_at")]
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 更新时间
    /// </summary>
    [JsonPropertyName("updated_at")]
    public DateTime UpdatedAt { get; set; }
}

/// <summary>
/// 玩家列表数据
/// </summary>
public class PlayersListData : PagedData<Player>
{
    /// <summary>
    /// 玩家列表
    /// </summary>
    [JsonPropertyName("players")]
    public List<Player> Players
    {
        get => Items;
        set => Items = value;
    }
}

/// <summary>
/// 玩家登录信息
/// </summary>
public class PlayerLoginInfo
{
    /// <summary>
    /// 玩家ID
    /// </summary>
    [JsonPropertyName("player_id")]
    public int? PlayerId { get; set; }

    /// <summary>
    /// 玩家名称
    /// </summary>
    [JsonPropertyName("name")]
    public string? Name { get; set; }

    /// <summary>
    /// IP地址
    /// </summary>
    [JsonPropertyName("ip")]
    public string? IP { get; set; }
}

/// <summary>
/// 登录验证数据
/// </summary>
public class ValidateLoginData
{
    /// <summary>
    /// 是否允许登录
    /// </summary>
    [JsonPropertyName("allowed")]
    public bool Allowed { get; set; }

    /// <summary>
    /// 玩家ID
    /// </summary>
    [JsonPropertyName("player_id")]
    public int? PlayerId { get; set; }

    /// <summary>
    /// 拒绝原因
    /// </summary>
    [JsonPropertyName("reason")]
    public string? Reason { get; set; }
}

/// <summary>
/// 单个玩家验证信息
/// </summary>
public class PlayerValidateInfo
{
    /// <summary>
    /// 玩家名
    /// </summary>
    [JsonPropertyName("player_name")]
    public string PlayerName { get; set; } = string.Empty;

    /// <summary>
    /// IP地址
    /// </summary>
    [JsonPropertyName("ip")]
    public string IP { get; set; } = string.Empty;

    /// <summary>
    /// 客户端版本
    /// </summary>
    [JsonPropertyName("client_version")]
    public string? ClientVersion { get; set; }

    /// <summary>
    /// 协议版本
    /// </summary>
    [JsonPropertyName("protocol_version")]
    public string? ProtocolVersion { get; set; }
}

/// <summary>
/// IP风险信息
/// </summary>
public class IPRiskInfo
{
    /// <summary>
    /// IP地址
    /// </summary>
    [JsonPropertyName("ip")]
    public string IP { get; set; } = string.Empty;

    /// <summary>
    /// 风险等级
    /// </summary>
    [JsonPropertyName("risk_level")]
    public string RiskLevel { get; set; } = string.Empty;

    /// <summary>
    /// 风险评分
    /// </summary>
    [JsonPropertyName("risk_score")]
    public int RiskScore { get; set; }

    /// <summary>
    /// 是否被封禁
    /// </summary>
    [JsonPropertyName("banned")]
    public bool Banned { get; set; }

    /// <summary>
    /// 风险描述
    /// </summary>
    [JsonPropertyName("description")]
    public string Description { get; set; } = string.Empty;
}

/// <summary>
/// 单个玩家验证结果
/// </summary>
public class PlayerValidateResult
{
    /// <summary>
    /// 玩家名
    /// </summary>
    [JsonPropertyName("player_name")]
    public string PlayerName { get; set; } = string.Empty;

    /// <summary>
    /// 是否允许登录
    /// </summary>
    [JsonPropertyName("allowed")]
    public bool Allowed { get; set; }

    /// <summary>
    /// 玩家ID
    /// </summary>
    [JsonPropertyName("player_id")]
    public int? PlayerId { get; set; }

    /// <summary>
    /// 拒绝原因
    /// </summary>
    [JsonPropertyName("reason")]
    public string? Reason { get; set; }

    /// <summary>
    /// 是否为新玩家
    /// </summary>
    [JsonPropertyName("newbie")]
    public bool Newbie { get; set; } = false;

    /// <summary>
    /// IP风险信息
    /// </summary>
    [JsonPropertyName("ip_risk")]
    public IPRiskInfo? IPRisk { get; set; }
}

/// <summary>
/// 玩家验证响应
/// </summary>
public class ValidateResponse
{
    /// <summary>
    /// 验证结果列表
    /// </summary>
    [JsonPropertyName("results")]
    public List<PlayerValidateResult> Results { get; set; } = new();

    /// <summary>
    /// 处理时间戳
    /// </summary>
    [JsonPropertyName("processed_at")]
    public long ProcessedAt { get; set; }
}
